### DISCUSSION TITLE
Flex growing outside of parent

### DISCUSSION THREAD

**Discussion by dinonovak** [tailwindlabs/tailwindcss#4884] [Help]
Hi,

I am trying to set up CSS for screen layout
https://play.tailwindcss.com/eH2su6wj61

In my case red flexes are growing larger that parrent?
also what is the way to prevent content in flexes from growing flexes?

Thanx,
D

**Comment by olets** [comment]
You're close!

One common pattern for getting intrinsic-height content followed by an element that fills the remaining height is a `flex-direction: column` parent with two children one of which has `flex-grow: 1`. https://codepen.io/henry/pen/WNjQWVB?editors=1100

Nest a couple of those to get your layout https://play.tailwindcss.com/5uQQd5Lx2h. There's one regression in my example: the "2/3"-width elements aren't actually twice the width of the 1/3-width elements. If it needs to be exact, drop the right margin and nest an additional level with right padding. Or use CSS Grid to do the widths.

For next time, it'll be easiest for people to help if the sample only includes things relevant to the question. See [How to create a Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) (for general CSS questions like this, you might even find there's already an answer on Stack Overflow!)
